exporter/trace/trace_proto.go

```diff
@@ -178,11 +180,36 @@ func protoFromReadOnlySpan(s sdktrace.ReadOnlySpan, defaultTraceAttributes []att
 		sp.TimeEvents.DroppedAnnotationsCount = clip32(droppedAnnotationsCount)
 	}
 
-	// TODO(ymotongpoo): add implementations for Links
+	sp.Links = linksProtoFromLinks(s.Links())
 
 	return sp
 }
 
+func linksProtoFromLinks(links []trace.Link) *tracepb.Span_Links {
+	var linksPb *tracepb.Span_Links
+
+	if numLinks := len(links); numLinks > 0 {
+		linksPb = &tracepb.Span_Links{}
+		numLinksToKeep := numLinks
+		if numLinksToKeep > maxNumLinks {
+			numLinksToKeep = maxNumLinks
+		}
+
+		for _, link := range links[:numLinksToKeep] {
+			linkPb := &tracepb.Span_Link{
+				TraceId: link.SpanContext.TraceID().String(),
+				SpanId:  link.SpanContext.SpanID().String(),
+				Type:    tracepb.Span_Link_TYPE_UNSPECIFIED,
+			}
+			copyAttributes(&linkPb.Attributes, link.Attributes)
+			linksPb.Link = append(linksPb.Link, linkPb)
+		}
+		linksPb.DroppedLinksCount = clip32(numLinks - numLinksToKeep)
+	}
+
+	return linksPb
+}
```

Comment:
Isn't it out of `linksProtoFromLinks`'s concern to drop extra links?
```suggestion
func linksProtoFromLinks(links []trace.Link) *tracepb.Span_Links {
	if len(links) == 0 {
		return nil
	}

	linksPb := &tracepb.Span_Links{
		Link: make([]*tracepb.Span_Link, 0, len(links)),
	}
	for _, link := range links {
		linkPb := &tracepb.Span_Link{
			TraceId: link.SpanContext.TraceID().String(),
			SpanId:  link.SpanContext.SpanID().String(),
			Type:    tracepb.Span_Link_TYPE_UNSPECIFIED,
		}
		copyAttributes(&linkPb.Attributes, link.Attributes)
		linksPb.Link = append(linksPb.Link, linkPb)
	}

	return linksPb
}
```

Reply:
Btw, I think ultimately we should remove all of the dropping logic from the exporter and pass [span limits](https://pkg.go.dev/go.opentelemetry.io/otel/sdk@v0.20.0/trace#WithSpanLimits) to the SDK in `NewExportPipeline()`.

Reply:
Well, the function name says it is supposed to convert span links to protobuf span links, not dropping them. What you see is not what you get something.

But, yes I agree not to drop anything unless it is explicitly there.

Reply:
@jsuereth made a good point offline that setting these limits will effect ALL exporters if the user has set up multiple pipelines. So I will keep the dropping and tweak this function name